packages/x-tailwindcss/src/x-tailwind-plugin/plugin.ts

```diff
@@ -38,22 +39,29 @@ export default plugin.withOptions(
       forEach(
         deepMerge({}, dynamicUtilities(helpers), options?.dynamicUtilities?.(helpers)),
         (key, { styles, values }) => {
-          helpers.matchUtilities({ [key]: styles }, { values: values ?? undefined });
+          helpers.matchUtilities(
+            { [key]: styles },
+            { respectPrefix: false, values: values ?? undefined }
+          );
         }
       );
-      helpers.addUtilities(deepMerge({}, utilities(helpers), options?.utilities?.(helpers)));
+      helpers.addUtilities(deepMerge({}, utilities(helpers), options?.utilities?.(helpers)), {
+        respectPrefix: false
+      });
 
       options?.extra?.(helpers);
       helpers.addVariant('selected', '&.selected');
     };
   },
-  function (options) {
+  function (options = {}) {
     return {
-      theme: deepMerge(
-        { colors: { current: 'currentColor', transparent: 'transparent' } },
-        theme,
-        options?.theme
-      )
-    } as Config;
+      theme: {
+        x: deepMerge(
```

Comment:
now we will have a 'x' property on the theme, so we are able there to customize the theme provided by x.
Only x classes will be affected when manipulating styles within the `theme.x` block. 
isolation++ 👍🏽

Reply:
You also need to reference the default `theme` of the plugin to allow Tailwind generates the property identifiers like `bg`, `text`, etc. with the default `theme`. The theme under `x` is only used to generate the custom properties in the plugin.
```TS
function (options = {}) {
  const customXTheme = deepMerge(
    { colors: { current: 'currentColor', transparent: 'transparent' } },
    xTheme,
    options.theme
  );
  return {
    theme: {
      extend: customXTheme,
      x: customXTheme
    }
  };
}
```

Remember to replace all `x-bg-...` for `bg-...` (some are still pending)

Reply:
if we add the customXTheme to the extend field... aren't we still overriding the default values from the theme with the ones from the x plugin? 🤔